ext/ffi/ffi.c

```diff
@@ -4469,6 +4469,318 @@ ZEND_METHOD(FFI_CType, getName) /* {{{ */
+	if (type->kind != ZEND_FFI_TYPE_ARRAY) {
+		zend_throw_error(zend_ffi_exception_ce, "FFI\\CType is not an array");
+		RETURN_THROWS();
+	}
+	RETURN_LONG(type->array.length);
+}
+/* }}} */
+
+ZEND_METHOD(FFI_CType, getPointerType) /* {{{ */
+{
+	zend_ffi_ctype *ctype = (zend_ffi_ctype*)(Z_OBJ_P(ZEND_THIS));
+	zend_ffi_ctype *ret;
+	zend_ffi_type *type;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	type = ZEND_FFI_TYPE(ctype->type);
+	if (type->kind != ZEND_FFI_TYPE_POINTER) {
+		zend_throw_error(zend_ffi_exception_ce, "FFI\\CType is not a pointer");
+		RETURN_THROWS();
+	}
+
+	ret = (zend_ffi_ctype*)zend_ffi_ctype_new(zend_ffi_ctype_ce);
+	ret->type = ZEND_FFI_TYPE(type->pointer.type);
+	RETURN_OBJ(&ret->std);
+}
+/* }}} */
+
+ZEND_METHOD(FFI_CType, getStructFieldNames) /* {{{ */
+{
+	zend_ffi_ctype *ctype = (zend_ffi_ctype*)(Z_OBJ_P(ZEND_THIS));
+	zend_ffi_type *type;
+	HashTable *ht;
+	zend_string* name;
+	zval zv;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	type = ZEND_FFI_TYPE(ctype->type);
+	if (type->kind != ZEND_FFI_TYPE_STRUCT) {
+		zend_throw_error(zend_ffi_exception_ce, "FFI\\CType is not a structure");
+		RETURN_THROWS();
+	}
+
+	ht = zend_new_array(zend_hash_num_elements(&type->record.fields));
+	RETVAL_ARR(ht);
+	ZEND_HASH_FOREACH_STR_KEY(&type->record.fields, name) {
+		ZVAL_STR_COPY(&zv, name);
+		zend_hash_next_index_insert_new(ht, &zv);
+	} ZEND_HASH_FOREACH_END();
+}
+/* }}} */
+
+ZEND_METHOD(FFI_CType, getStructFieldOffset) /* {{{ */
+{
+	zend_ffi_ctype *ctype = (zend_ffi_ctype*)(Z_OBJ_P(ZEND_THIS));
+	zend_ffi_type *type;
+	zend_string *name;
+	zend_ffi_field *ptr;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_STR(name)
+	ZEND_PARSE_PARAMETERS_END();
+
+	type = ZEND_FFI_TYPE(ctype->type);
+	if (type->kind != ZEND_FFI_TYPE_STRUCT) {
+		zend_throw_error(zend_ffi_exception_ce, "FFI\\CType is not a structure");
+		RETURN_THROWS();
+	}
+
+	ptr = zend_hash_find_ptr(&type->record.fields, name);
+	if (!ptr) {
+		zend_throw_error(zend_ffi_exception_ce, "Wrong fileld name");
+		RETURN_THROWS();
+	}
+	RETURN_LONG(ptr->offset);
+}
+/* }}} */
+
+ZEND_METHOD(FFI_CType, getStructFieldType) /* {{{ */
+{
+	zend_ffi_ctype *ctype = (zend_ffi_ctype*)(Z_OBJ_P(ZEND_THIS));
+	zend_ffi_type *type;
+	zend_string *name;
+	zend_ffi_field *ptr;
+	zend_ffi_ctype *ret;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_STR(name)
+	ZEND_PARSE_PARAMETERS_END();
+
+	type = ZEND_FFI_TYPE(ctype->type);
+	if (type->kind != ZEND_FFI_TYPE_STRUCT) {
+		zend_throw_error(zend_ffi_exception_ce, "FFI\\CType is not a structure");
+		RETURN_THROWS();
+	}
+
+	ptr = zend_hash_find_ptr(&type->record.fields, name);
+	if (!ptr) {
+		zend_throw_error(zend_ffi_exception_ce, "Wrong fileld name");
+		RETURN_THROWS();
+	}
+
+	ret = (zend_ffi_ctype*)zend_ffi_ctype_new(zend_ffi_ctype_ce);
+	ret->type = ZEND_FFI_TYPE(ptr->type);
+	RETURN_OBJ(&ret->std);
+}
+/* }}} */
+
+ZEND_METHOD(FFI_CType, getFuncABI) /* {{{ */
```

Comment:
I think keeping ABI is better here, because it's a term of art. If you really want to use a different term `CallingConvention` could work as well, but `CallType` doesn't seem correct in this context.

Reply:
Ok, let's keep getFunctionABI then if you don't have objections. I also like `getFunctionCallingConvention()` - this will be readable.